File: Server-CSharp/Server-CSharp/YT/Connection.cs
using Google.Protobuf;
using System;
using System.Collections.Generic;
using System.Text;
using TKcp;

namespace YT
{
    public class Connection
    {
        //玩家
        public Player player;
        //客户端的Peer
        public uint Conv;
        //缓存区
        public ByteArray readBuff = new ByteArray();

        public Server server;

        public Connection(uint conv) {
            Conv = conv;
        }

        public void Send(IMessage message) {
            server.Send(Conv, message);
        }
    }
}

File: Server-Go/YT/Connection.go
package YT

import (
	"ReaminHistory/YT/Base"
	"google.golang.org/protobuf/runtime/protoiface"
)

type Connection struct {
	//客户端Peer
	Conv uint32
	//缓存区
	readBuf *Base.ByteArray
	Server  *Server
}

func NewConnection(conv uint32) *Connection {
	return &Connection{Conv: conv}
}

func (connection *Connection) Send(message protoiface.MessageV1) {
	connection.Server.Send(connection.Conv, message)
}
